### PR TITLE
docs: add production checklist

### PR DESCRIPTION
**Pull request by thesephirot**
I doublechecked some OS production checklists and I think a generic short list is IMO better suited as the individual setups may/will vary quite heavily.

@fforootd 
@eliobischof 

Did I miss something?

### CODE REVIEW

**Comment by fforootd** [comment]
You are aware that we have https://zitadel.com/docs/self-hosting/manage/production , right?

It looks to me that we should integrate the content into that page.

Besides that, can you please use a semrel title and set this PR to "ready for review", ty!

**Review by fforootd — COMMENTED** [review]
More to come as soon as i have time again, but that and my comment [here ](https://github.com/zitadel/zitadel/pull/4964#issuecomment-1368851023) is what i can provide so far

File: docs/docs/self-hosting/manage/production.md [hunk 1]
```diff
@@ -135,3 +135,63 @@ DefaultInstance:
...
+
+## Checklist
+To ease your production setup we created a step by step best practise checklist you may wish to follow.
+
+### infrastructure configuration
+
+- [] make use of configmanagement such as Terraform to provision all of the below
+- [] use version control to store the provisioning
+- [] use a secrets manager to save your sensible informations
+- [] reduce the manual interaction with your platform to an absolute minimum
```
[the remark this answers is not in the document]
Comment (thesephirot) [reply docs/docs/self-hosting/manage/production.md:150]: thats what I did, I find it ugly as well, I will doublecheck formatting anyways

[the remark this answers is not in the document]
Comment (fforootd) [reply docs/docs/self-hosting/manage/production.md:150]: In MD only with spaces should work if i remember correctly

See this test below:

- [ ] Test
- [] Test

[the remark this answers is not in the document]
Comment (thesephirot) [reply docs/docs/self-hosting/manage/production.md]: a space on the correct place, quite makes a difference, my bad. looks better now.

**Comment by thesephirot** [comment]
> 

as for the redundancy, I am with you but don't want to rip the list into pieces. I will try to consolidate or put the list separately

**Comment by thesephirot** [comment]
> ~You are aware that we have https://zitadel.com/docs/self-hosting/manage/production , right?~ ~It looks to me that we should integrate the content into that page.~
> 
> My preview was broken and did not show the same file... mybad 😁
> 
> Some thoughts/observations of mine for this:
> 
> * I we want to have one document we should rename the title. It make a little sense to me having a production checklist with a checklist in it.
> * Checklist boxes look broken in the preview https://deploy-preview-4964--docs-zitadel-com.netlify.app/docs/self-hosting/manage/production#checklist
> * Some of the content feels redundant since it is in text and checkboxes, I am not sure how we can reduce this
> * Besides that, can you please use a semrel title and set this PR to "ready for review", ty!

I had a play around. To me the best solution is to completely separate the checklist from the Production setup page. Sure there is some redundancy, but that is okay im my opinion as some steps might need further explanation on the "setup page".

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  - [ ] secure database connections from outside your network and/or use an internal subnet for database connectivity
+- [ ] High Availability for critical infrastructure components (depending on your setup)
+  - [ ] Loadbalancer
+  - [ ] [Reverse Proxies](https://zitadel.com/docs/self-hosting/manage/reverseproxy/reverse_proxy)
+  - [ ] Web Application Firewall
+
+#### Networking
+- [ ] Use a Layer 7 Web Application Firewall to secure ZITADEL that supports **[HTTP/2](/docs/self-hosting/manage/http2)**
+  - [ ] secure the access by IP if needed
+  - [ ] secure the access by [ratelimits](https://zitadel.com/docs/apis/ratelimits) for specific endpoints (e.g. API vs frontend) to secure availability on high load
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:32]:
```suggestion
  - [ ] secure the access by rate limits for specific endpoints (e.g. API vs frontend) to secure availability on high load. See the [ZITADEL Cloud rate limits](https://zitadel.com/docs/apis/ratelimits) for reference.
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  - [ ] doublecheck your Firewall for IPv6 connectivity and change accordingly
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:33]:
```suggestion
  - [ ] doublecheck your firewall for IPv6 connectivity and change accordingly
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+---
+title: Production Checklist
+---
+
+
+To apply best practices to your production setup we created a step by step checklist you may wish to follow.
+
+### infrastructure configuration
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:8]:
```suggestion
### Infrastructure Configuration
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+- [ ] make use of configmanagement such as Terraform to provision all of the below
+- [ ] use version control to store the provisioning
+- [ ] use a secrets manager to save your sensible informations
+- [ ] reduce the manual interaction with your platform to an absolute minimum 
+#### HA Setup
+- [ ] High Availability for ZITADEL containers
+  - [ ] use container orchestrator such as Kubernetes or
+  - [ ] use serverless architecture such as Knative or a hyperscaler equivalent (e.g. CloudRun from Google)
+  - [ ] separate `zitadel init` and `zitadel setup` for fast startup times when scaling ZITADEL
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:18]: Add link to the updating and scaling page

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  - [ ] secure the access by [ratelimits](https://zitadel.com/docs/apis/ratelimits) for specific endpoints (e.g. API vs frontend) to secure availability on high load
+  - [ ] doublecheck your Firewall for IPv6 connectivity and change accordingly
+
+### ZITADEL configuration
+- [ ] configure a valid [SMTP Server](https://zitadel.com/docs/guides/manage/console/instance-settings#smtp) and test emails
+- [ ] Add [Custom Branding](https://zitadel.com/docs/guides/manage/customize/branding) if required
+- [ ] configure a valid [SMS Service](https://zitadel.com/docs/guides/manage/console/instance-settings#sms) such as Twilio if needed
+- [ ] configure your privacy policy, terms of service and a help Link if needed
+- [ ] secure your [masterkey](https://zitadel.com/docs/self-hosting/manage/configure)
+
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:41]: - declare and apply zitadel configuration using the zitadel terraform provider (link)

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+#### Networking
+- [ ] Use a Layer 7 Web Application Firewall to secure ZITADEL that supports **[HTTP/2](/docs/self-hosting/manage/http2)**
+  - [ ] secure the access by IP if needed
+  - [ ] secure the access by [ratelimits](https://zitadel.com/docs/apis/ratelimits) for specific endpoints (e.g. API vs frontend) to secure availability on high load
+  - [ ] doublecheck your Firewall for IPv6 connectivity and change accordingly
+
+### ZITADEL configuration
+- [ ] configure a valid [SMTP Server](https://zitadel.com/docs/guides/manage/console/instance-settings#smtp) and test emails
+- [ ] Add [Custom Branding](https://zitadel.com/docs/guides/manage/customize/branding) if required
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:37]:
```suggestion
- [ ] add [custom branding](https://zitadel.com/docs/guides/manage/customize/branding) if required
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  - [ ] Web Application Firewall
+
+#### Networking
+- [ ] Use a Layer 7 Web Application Firewall to secure ZITADEL that supports **[HTTP/2](/docs/self-hosting/manage/http2)**
+  - [ ] secure the access by IP if needed
+  - [ ] secure the access by [ratelimits](https://zitadel.com/docs/apis/ratelimits) for specific endpoints (e.g. API vs frontend) to secure availability on high load
+  - [ ] doublecheck your Firewall for IPv6 connectivity and change accordingly
+
+### ZITADEL configuration
+- [ ] configure a valid [SMTP Server](https://zitadel.com/docs/guides/manage/console/instance-settings#smtp) and test emails
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:36]:
```suggestion
- [ ] configure a valid [SMTP server](https://zitadel.com/docs/guides/manage/console/instance-settings#smtp) and test emails
```

File: docs/docs/self-hosting/manage/production.md [hunk 9]
```diff
@@ -1,10 +1,14 @@
...
-title: Production Checklist
+title: Production Setup
 ---
 
 As soon as you successfully deployed ZITADEL as a proof of concept using one of our [deployment guides](/docs/self-hosting/deploy/overview),
 you are ready to configure ZITADEL for production usage.
 
+
+## TL;DR
+We created a [Production Checklist](./productionchecklist.md) as a technical step by step guide.
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/production.md:10]:
```suggestion
We created a [production checklist](./productionchecklist.md) as a technical step by step guide.
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+To apply best practices to your production setup we created a step by step checklist you may wish to follow.
+
+### infrastructure configuration
+
+- [ ] make use of configmanagement such as Terraform to provision all of the below
+- [ ] use version control to store the provisioning
+- [ ] use a secrets manager to save your sensible informations
+- [ ] reduce the manual interaction with your platform to an absolute minimum 
+#### HA Setup
+- [ ] High Availability for ZITADEL containers
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:15]:
```suggestion
- [ ] high availability for ZITADEL containers
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+  - [ ] use container orchestrator such as Kubernetes or
+  - [ ] use serverless architecture such as Knative or a hyperscaler equivalent (e.g. CloudRun from Google)
+  - [ ] separate `zitadel init` and `zitadel setup` for fast startup times when scaling ZITADEL
+- [ ] High Availability for database
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:19]:
```suggestion
- [ ] high availability for database 
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+- [ ] use version control to store the provisioning
+- [ ] use a secrets manager to save your sensible informations
+- [ ] reduce the manual interaction with your platform to an absolute minimum 
+#### HA Setup
+- [ ] High Availability for ZITADEL containers
+  - [ ] use container orchestrator such as Kubernetes or
+  - [ ] use serverless architecture such as Knative or a hyperscaler equivalent (e.g. CloudRun from Google)
+  - [ ] separate `zitadel init` and `zitadel setup` for fast startup times when scaling ZITADEL
+- [ ] High Availability for database 
+  - [ ] follow the [Production Checklist](https://www.cockroachlabs.com/docs/stable/recommended-production-settings.html) for CockroachDB if you selfhost the database or use [CockroachDB cloud](https://www.cockroachlabs.com/docs/cockroachcloud/create-an-account.html)
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:20]:
```suggestion
  - [ ] follow the [production checklist](https://www.cockroachlabs.com/docs/stable/recommended-production-settings.html) for CockroachDB if you selfhost the database or use [CockroachDB Cloud](https://www.cockroachlabs.com/docs/cockroachcloud/create-an-account.html)
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+  - [ ] configure backups on a regular basis for the Database
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:21]:
```suggestion
  - [ ] configure backups on a regular basis for the database
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+  - [ ] test a restore scenario before going live
+  - [ ] secure database connections from outside your network and/or use an internal subnet for database connectivity
+- [ ] High Availability for critical infrastructure components (depending on your setup)
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:24]:
```suggestion
- [ ] high availability for critical infrastructure components (depending on your setup)
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+  - [ ] Loadbalancer
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:25]:
```suggestion
  - [ ] loadbalancer
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+  - [ ] [Reverse Proxies](https://zitadel.com/docs/self-hosting/manage/reverseproxy/reverse_proxy)
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:26]:
```suggestion
  - [ ] [reverse proxies](https://zitadel.com/docs/self-hosting/manage/reverseproxy/reverse_proxy)
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+  - [ ] Web Application Firewall
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:27]:
```suggestion
  - [ ] web application firewall
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+
+#### Networking
+- [ ] Use a Layer 7 Web Application Firewall to secure ZITADEL that supports **[HTTP/2](/docs/self-hosting/manage/http2)**
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:30]:
```suggestion
- [ ] use a layer 7 web application firewall to secure ZITADEL that supports **[HTTP/2](/docs/self-hosting/manage/http2)**
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+  - [ ] secure the access by IP if needed
+  - [ ] secure the access by [ratelimits](https://zitadel.com/docs/apis/ratelimits) for specific endpoints (e.g. API vs frontend) to secure availability on high load
+  - [ ] doublecheck your Firewall for IPv6 connectivity and change accordingly
+
+### ZITADEL configuration
+- [ ] configure a valid [SMTP Server](https://zitadel.com/docs/guides/manage/console/instance-settings#smtp) and test emails
+- [ ] Add [Custom Branding](https://zitadel.com/docs/guides/manage/customize/branding) if required
+- [ ] configure a valid [SMS Service](https://zitadel.com/docs/guides/manage/console/instance-settings#sms) such as Twilio if needed
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:38]:
```suggestion
- [ ] configure a valid [SMS service](https://zitadel.com/docs/guides/manage/console/instance-settings#sms) such as Twilio if needed
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+- [ ] configure your privacy policy, terms of service and a help Link if needed
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:39]:
```suggestion
- [ ] configure your privacy policy, terms of service and a help link if needed
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+- [ ] secure your [masterkey](https://zitadel.com/docs/self-hosting/manage/configure)
+
+### Security
+- [ ] use a FQDN and a trusted valid certificate for external [TLS](https://zitadel.com/docs/self-hosting/manage/tls_modes#http2) connections
+- [ ] make use of different Serviceaccounts to secure ZITADEL within your Hyperscaler or Kubernetes
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:44]:
```suggestion
- [ ] make use of different service accounts to secure ZITADEL within your hyperscaler or Kubernetes 
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+- [ ] configure a valid [SMTP Server](https://zitadel.com/docs/guides/manage/console/instance-settings#smtp) and test emails
+- [ ] Add [Custom Branding](https://zitadel.com/docs/guides/manage/customize/branding) if required
+- [ ] configure a valid [SMS Service](https://zitadel.com/docs/guides/manage/console/instance-settings#sms) such as Twilio if needed
+- [ ] configure your privacy policy, terms of service and a help Link if needed
+- [ ] secure your [masterkey](https://zitadel.com/docs/self-hosting/manage/configure)
+
+### Security
+- [ ] use a FQDN and a trusted valid certificate for external [TLS](https://zitadel.com/docs/self-hosting/manage/tls_modes#http2) connections
+- [ ] make use of different Serviceaccounts to secure ZITADEL within your Hyperscaler or Kubernetes 
+- [ ] make use of a CDN Service if needed to ease maintainability and Firewall/DNS/WAF configuration
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:45]:
```suggestion
- [ ] make use of a CDN service if needed to ease maintainability and firewall/DNS/WAF configuration
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+- [ ] make use of a [security scanner](https://owasp.org/www-community/Vulnerability_Scanning_Tools) to test your application and cluster
+
+### Monitoring
+Use a appropriate Monitoring Solution to have an overview about your ZITADEL instance. In particular you may want to watch out for things like:
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:49]:
```suggestion
Use an appropriate monitoring solution to have an overview about your ZITADEL instance. In particular you may want to watch out for things like:
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
+
+- [ ] CPU and Memory of ZITADEL and the database
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:51]:
```suggestion
- [ ] CPU and memory of ZITADEL and the database
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 25]
```diff
@@ -0,0 +1,59 @@
...
+- [ ] open database connections
+- [ ] running instances of ZITADEL and the database
+- [ ] latency of requests
+- [ ] requests per second
+- [ ] requests by URL/endpoint
+- [ ] lifetime of TLS certificates
+- [ ] ZITADEL and database logs
+- [ ] ZITADEL [Metrics](https://zitadel.com/docs/apis/observability/metrics)
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:59]:
```suggestion
- [ ] ZITADEL [metrics](https://zitadel.com/docs/apis/observability/metrics)
```

**Review by eliobischof — CHANGES_REQUESTED** [review]

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 26]
```diff
@@ -0,0 +1,60 @@
...
+  - [ ] Web Application Firewall
+
+#### Networking
+- [ ] Use a Layer 7 Web Application Firewall to secure ZITADEL that supports **[HTTP/2](/docs/self-hosting/manage/http2)**
+  - [ ] secure the access by IP if needed
+  - [ ] secure the access by rate limits for specific endpoints (e.g. API vs frontend) to secure availability on high load. See the [ZITADEL Cloud rate limits](https://zitadel.com/docs/apis/ratelimits) for reference.
+  - [ ] doublecheck your firewall for IPv6 connectivity and change accordingly
+
+### ZITADEL configuration
+- [ ] configure a valid [SMTP Server](https://zitadel.com/docs/guides/manage/console/instance-settings#smtp) and test emails
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:36]:
```suggestion
- [ ] configure a valid [SMTP Server](/docs/guides/manage/console/instance-settings#smtp) and test emails
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 27]
```diff
@@ -0,0 +1,60 @@
...
+- [ ] Add [Custom Branding](https://zitadel.com/docs/guides/manage/customize/branding) if required
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:37]:
```suggestion
- [ ] Add [Custom Branding](/docs/guides/manage/customize/branding) if required
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+- [ ] configure a valid [SMS Service](https://zitadel.com/docs/guides/manage/console/instance-settings#sms) such as Twilio if needed
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:38]:
```suggestion
- [ ] configure a valid [SMS Service](/docs/guides/manage/console/instance-settings#sms) such as Twilio if needed
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 29]
```diff
@@ -0,0 +1,60 @@
...
+- [ ] configure your privacy policy, terms of service and a help Link if needed
+- [ ] secure your [masterkey](https://zitadel.com/docs/self-hosting/manage/configure)
+- [ ] declare and apply zitadel configuration using the zitadel terraform [provider](https://github.com/zitadel/terraform-provider-zitadel) 
+
+### Security
+- [ ] use a FQDN and a trusted valid certificate for external [TLS](https://zitadel.com/docs/self-hosting/manage/tls_modes#http2) connections
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:44]:
```suggestion
- [ ] use a FQDN and a trusted valid certificate for external [TLS](/docs/self-hosting/manage/tls_modes#http2) connections
```

File: docs/docs/self-hosting/manage/productionchecklist.md [hunk 30]
```diff
@@ -0,0 +1,60 @@
...
+
+- [ ] CPU and memory of ZITADEL and the database
+- [ ] open database connections
+- [ ] running instances of ZITADEL and the database
+- [ ] latency of requests
+- [ ] requests per second
+- [ ] requests by URL/endpoint
+- [ ] lifetime of TLS certificates
+- [ ] ZITADEL and database logs
+- [ ] ZITADEL [metrics](https://zitadel.com/docs/apis/observability/metrics)
```
Comment (eliobischof) [inline docs/docs/self-hosting/manage/productionchecklist.md:60]:
```suggestion
- [ ] ZITADEL [metrics](/docs/apis/observability/metrics)
```

**Review by eliobischof — CHANGES_REQUESTED** [review]

**Review by livio-a — APPROVED** [review]

File: docs/docs/self-hosting/manage/productionchecklist.md (hunk 19 above)
Comment (eliobischof) [reply docs/docs/self-hosting/manage/productionchecklist.md]: ?

File: docs/docs/self-hosting/manage/productionchecklist.md (hunk 18 above)
Comment (eliobischof) [reply docs/docs/self-hosting/manage/productionchecklist.md:30]: ?

**Pull request merged**